index.d.ts

```diff
@@ -14,20 +14,82 @@ export function isProbablyReaderable(
   }
 ): boolean;
 
+export interface ReadabilityOptions {
+  /**
+   * Whether to output debug messages. Defaults to `false`.
+   */
+  debug?: boolean;
+  /**
+   * The maximum number of elements to parse. If the document exceeds this,
+   * Readability will stop processing. Useful for performance on very large documents.
+   * Defaults to 0 (no limit).
+   */
+  maxElemsToParse?: number;
+  /**
+   * The number of top candidate nodes to consider when determining the main article content.
+   * A higher number might lead to better results but could increase processing time.
+   * Defaults to 5.
+   */
+  nbTopCandidates?: number;
+  /**
+   * The minimum number of characters required for a text node to be considered
+   * significant and included in the article content.
+   * Defaults to 500.
+   */
+  charThreshold?: number;
+  /**
+   * An array of class names to preserve. If `keepClasses` is `true`,
+   * only classes in this array will be kept. Defaults to an empty array.
+   */
+  classesToPreserve?: string[];
+  /**
+   * If `true`, Readability will retain the original class names of elements
+   * in the parsed article content. If `classesToPreserve` is also set,
+   * only those specified classes will be kept. Defaults to `false`.
+   */
+  keepClasses?: boolean;
+  /**
+   * A function that serializes an HTML element into a string.
+   * Defaults to `el => el.innerHTML`. This is used to get the content
+   * of the parsed article.
+   * @param el The Node to serialize.
+   * @returns The HTML string representation of the element's content.
+   */
+  serializer?: (el: Node) => string;
+  /**
+   * If `true`, Readability will not attempt to parse or extract
+   * JSON-LD structured data from the document. Defaults to `false`.
+   */
+  disableJSONLD?: boolean;
+  /**
+   * A regular expression used to validate video URLs. Only videos
+   * matching this regex will be included in the parsed content.
+   * Defaults to a regex that allows common video embedding platforms.
+   */
+  allowedVideoRegex?: RegExp;
+  /**
+   * A modifier applied to the link density score of an element.
+   * This influences how Readability judges the main content area,
+   * potentially helping with documents that have many or few links.
+   * Defaults to 1.
+   */
+  linkDensityModifier?: number;
+  /**
+   * An object of callback functions that can be used to extend Readability's behavior.
+   * Currently, no specific callbacks are detailed or widely used.
+   * It's treated as a generic object (`Record<string, unknown>`).
+   */
+  callbacks?: Record<string, unknown>;
```

Comment:
This doesn't seem to exist?

Reply:
Isn’t it what’s here https://github.com/mozilla/readability/blob/main/test/test-pages/ehow-2/source.html#L982 ?

Reply:
Right, the JS there has nothing to do with the module.

Reply:
(it's in the `test-pages` directory which is just test pages from the internet; JS there doesn't interact with the module in any way)